server/src/test/java/com/kentaromatt/todoapi/ToDoControllerTest.java

```diff
@@ -230,4 +231,77 @@ public void testUpdateSetsIncompleteToIncomplete() throws Exception {
         assertNotNull(foundToDo);
         assertFalse(foundToDo.getIsComplete());
     }
+
+    @Test
+    public void testPostToToDosReturns200OnSuccess() throws Exception {
+        Map<String, String> requestBodyObject = Map.of("description", "Feed cat", "dueDate", "2021-04-20");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String requestBodyJson = objectMapper.writeValueAsString(requestBodyObject);
+        
+        mvc.perform(
+            MockMvcRequestBuilders.post("/api/todos")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBodyJson))
+                    .andExpect(MockMvcResultMatchers.status().is2xxSuccessful());
+    }
+
+    @Test
+    public void testPostToToDosReturnsJson() throws Exception {
+        Map<String, String> requestBodyObject = Map.of("description", "Walk cat", "dueDate", "2021-04-20");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String requestBodyJson = objectMapper.writeValueAsString(requestBodyObject);
+        
+        mvc.perform(
+            MockMvcRequestBuilders.post("/api/todos")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBodyJson))
+                    .andExpect(MockMvcResultMatchers.jsonPath("$.dueDate", equalTo("2021-04-20")))
+                    .andExpect(MockMvcResultMatchers.jsonPath("$.description", equalTo("Walk cat")))
+                    .andExpect(MockMvcResultMatchers.jsonPath("$.isComplete", equalTo("false")))
+                    .andExpect(MockMvcResultMatchers.jsonPath("$.id").exists());
+    }
+
+    @Test
+    public void testPostToToDosReturns400IfMissingDescription() throws Exception {
+        Map<String, String> requestBodyObject = Map.of("dueDate", "2021-04-20");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String requestBodyJson = objectMapper.writeValueAsString(requestBodyObject);
+        
+        mvc.perform(
+            MockMvcRequestBuilders.post("/api/todos")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBodyJson))
+                    .andExpect(MockMvcResultMatchers.status().is4xxClientError());
+    }
+
+    @Test
+    public void testPostToToDosReturns400IfDescriptionIsEmptyString() throws Exception {
+        Map<String, String> requestBodyObject = Map.of("dueDate", "2021-04-20");
```

Comment:
it says "description is empty string" in the method name, but isn't the description `null`? did you mean to add `"description", ""` inside `Map.of(...)`?

Reply:
Thanks for catching - fixed 👍